src/HDWallet.cpp

```diff
@@ -120,34 +123,35 @@ DerivationPath HDWallet::cardanoStakingDerivationPath(const DerivationPath& path
 PrivateKey HDWallet::getKey(TWCoinType coin, const DerivationPath& derivationPath) const {
     const auto curve = TWCoinTypeCurve(coin);
     const auto privateKeyType = getPrivateKeyType(curve);
-    const auto node = getNode(*this, curve, derivationPath);
+    auto node = getNode(*this, curve, derivationPath);
     switch (privateKeyType) {
-        case PrivateKeyTypeDoubleExtended: // special handling for Cardano
-            {
-                if (derivationPath.indices.size() < 4 || derivationPath.indices[3].value > 1) {
-                    // invalid derivation path
-                    return PrivateKey(Data(192));
-                }
-                const DerivationPath stakingPath = cardanoStakingDerivationPath(derivationPath);
-
-                auto pkData = Data(node.private_key, node.private_key + PrivateKey::size);
-                auto extData = Data(node.private_key_extension, node.private_key_extension + PrivateKey::size);
-                auto chainCode = Data(node.chain_code, node.chain_code + PrivateKey::size);
-
-                // repeat with staking path
-                const auto node2 = getNode(*this, curve, stakingPath);
-                auto pkData2 = Data(node2.private_key, node2.private_key + PrivateKey::size);
-                auto extData2 = Data(node2.private_key_extension, node2.private_key_extension + PrivateKey::size);
-                auto chainCode2 = Data(node2.chain_code, node2.chain_code + PrivateKey::size);
-
-                return PrivateKey(pkData, extData, chainCode, pkData2, extData2, chainCode2);
-            }
-
-        case PrivateKeyTypeDefault32:
-        default:
-            // default path
-            auto data = Data(node.private_key, node.private_key + PrivateKey::size);
-            return PrivateKey(data);
+    case PrivateKeyTypeDoubleExtended: {
```

Comment:
A comment here with 'Cardano' may be useful when navigating code; can we keep it?

Reply:
I will rename it `PrivateKeyTypeCardano`, it's not common extended 25519 key and I found it's misleading, because it contains two keys